Nos/NosApp.swift

```diff
@@ -33,15 +33,19 @@ struct NosApp: App {
                     await persistenceController.cleanupEntities()
                 }
                 .onChange(of: scenePhase) { _, newPhase in
-                    // TODO: save all contexts, not just the view and background.
-                    if newPhase == .inactive {
+                    switch newPhase {
+                    case .inactive:
                         Log.info("Scene change: inactive")
-                        try? persistenceController.saveAll()
-                    } else if newPhase == .active {
+                    case .active:
                         Log.info("Scene change: active")
-                    } else if newPhase == .background {
+                    case .background:
                         Log.info("Scene change: background")
-                        try? persistenceController.saveAll()
+                        Task {
+                            // TODO: save all contexts, not just the view and background.
+                            try await persistenceController.saveAll()
```

Comment:
The app always passed through "inactive" before going to the background, so we were calling `saveAll()` too much. The docs say that the app will always go into the background before being closed, so it seems safe to only save there.